flow/local.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @flow
+ */
+
+type TAXObjectQueryMap<E, K, V> = {
```

Comment:
@ljharb what do you think about this interface to re-introduce iterator support for https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/814? It's a bit of a mashup of your Array.prototype[.keys,.values,.entries] modules.

Reply:
Seems good - but I’m not entirely sure why this is better than a simple array - we could even export both an array and a set, or both an object and a map?

Reply:
I originally went with a `Map` because in some case the keys in these maps are objects. So the `entries` method returns an array of tuples. And the `get` method becomes a convenience method for accessing values via complex keys. A straight array of the tuples would work as a top-level return value in the module, but it puts a lot of responsibility on the implementor. So the goal of this whole endeavor is to reproduce the required methods of a `Map` where `Map` is supported. In a way, it's a bonus that the `set`, `delete` and `clear` methods aren't implemented because it makes the module's data immutable in practice.

Reply:
in that case this seems fine; it's basically a "Map-like".